packages/core/src/operator/operator-runtime/validateConfirmData.ts

```diff
@@ -0,0 +1,144 @@
+
+    try {
+        if (event && currentChallenge.rollupUsed === config.rollupAddress) {
+
+            const currentAssertionId = Number(currentChallenge.assertionId);                // Destructure Current Assertion ID     
+            let assertionIds: number[] = [];                                                // Create an array to store the assertion IDs  
+
+            // Loop through the assertion IDs and add them to the array
+            for (let id = Number(operatorState.previousChallengeAssertionId) + 1; id <= currentAssertionId; id++) {
+                assertionIds.push(id);
+            }
+
+            const isBatch = assertionIds.length > 1;                                        // Check if the challenge is a batch or single challenge        
+            let confirmDataList: string[] = [];                                             // Create an array to store the confirm data for each assertionId                          
+
+            if (isBatch) {
+                const { confirmData } = await getConfirmDataAndHash(assertionIds, subgraphIsHealthy);
+                confirmDataList = confirmData;                                              // Set the confirm data list        
+            } else {
+                confirmDataList = [currentChallenge.assertionStateRootOrConfirmData];       // Set the initial confirm data assuming a single challenge
+            }
+
+            const errors = [];
+
+            // Validate the confirm data for each assertionId
+            for (let i = 0; i < confirmDataList.length; i++) {
+                const confirmData = confirmDataList[i];
+                const assertionId = assertionIds[i];
+
+                try {
+                    const { publicNodeBucket, error } = await compareWithCDN(assertionId, confirmData);
+
+                    if (error) {
+                        errors.push({ assertionId, error });
+                        operatorState.onAssertionMissMatchCb(publicNodeBucket, currentChallenge, error);
+                    }else{
+                        operatorState.cachedLogger(`Comparison between PublicNode and Challenger was successful for assertion ${assertionId}.`);
+                    }
+
+                } catch (error) {
+                    const errorMessage = error instanceof Error ? error.message : 'An unknown error occurred';
+                    errors.push({ assertionId, error: errorMessage });
+                    operatorState.onAssertionMissMatchCb(undefined, currentChallenge, errorMessage);
+                }
+            }
+
+            // Log all errors together
+            if (errors.length > 0) {
+                const errorLog = errors.map(e => `Assertion ${e.assertionId}: ${e.error}`).join('\n');
+                operatorState.cachedLogger(`Encountered errors during validation:\n${errorLog}`);
```

Comment:
We should call `onAssertionMissMatchCb` here once for all the errors.

Reply:
This has been revised, thank you.